### PR TITLE
Pagination improvement

### PR DESCRIPTION
**Pull request by adudekyeti**
Pagination improvement

### CODE REVIEW

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 1]
```diff
@@ -875,177 +876,142 @@ jQuery.Class("Vtiger_List_Js", {
...
+	/**
+	 * Jump to next page
+	 */
+	jumpToNextPage(e) {
+		const self = this,
+			listViewPageDiv = this.getListViewContainer();
+		let aDeferred = $.Deferred(),
+			element = $(e.currentTarget),
+			pageNumber = listViewPageDiv.find('#pageNumber');
+		if (listViewPageDiv.find(element).hasClass('disabled')) {
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]:
```suggestion
		if (element.hasClass('disabled')) {
```

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 2]
```diff
@@ -875,177 +876,142 @@ jQuery.Class("Vtiger_List_Js", {
...
+			return;
+		}
+		if (listViewPageDiv.find('#noOfEntries').val() === listViewPageDiv.find('#pageLimit').val()) {
+			let urlParams = {
+					orderby: listViewPageDiv.find('#orderBy').val(),
+					sortorder: listViewPageDiv.find("#sortOrder").val(),
+					viewname: self.getCurrentCvId()
+				},
+				nextPageNumber = parseInt(parseFloat(pageNumber.val())) + 1;
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]: **parseInt**(**parseFloat**(pageNumber.val())) ?

File: public_html/layouts/basic/modules/Settings/Vtiger/resources/List.js [hunk 3]
```diff
@@ -43,9 +43,9 @@ Vtiger_List_Js("Settings_Vtiger_List_Js", {
 	/*
 	 * Function to register the list view container
 	 */
-	getListViewContainer: function () {
+	getListViewContainer() {
 		if (this.listViewContainer == false) {
-			this.listViewContainer = jQuery('div.listViewContentDiv');
+			this.listViewContainer = $('div.listViewPageDiv');
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Settings/Vtiger/resources/List.js]: ?

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 4]
```diff
@@ -875,177 +876,142 @@ jQuery.Class("Vtiger_List_Js", {
...
+	 * @param {jQuery} element
+	 */
+	jumpToClickedPage(element) {
+		if (element.hasClass('disabled')) {
+			return false;
+		}
+		const self = this,
+			listViewPageDiv = this.getListViewContainer();
+		let pageNumberData = element.data("id"),
+			urlParams = {
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]: ?

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 5]
```diff
@@ -875,177 +876,119 @@ jQuery.Class("Vtiger_List_Js", {
...
+		let aDeferred = $.Deferred(),
+			urlParams = {
+				orderby: listViewPageDiv.find('#orderBy').val(),
+				sortorder: listViewPageDiv.find('#sortOrder').val(),
+				viewname: self.getCurrentCvId()
+			},
+			pageNumber = listViewPageDiv.find('#pageNumber');
+		pageNumber.val(page);
+		listViewPageDiv.find('.js-page-jump').val(page);
+		self.getListViewRecords(urlParams).done(function (data) {
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]:
```suggestion
		self.getListViewRecords({
		...
		}).done(function (data) {
```

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 6]
```diff
@@ -875,177 +876,119 @@ jQuery.Class("Vtiger_List_Js", {
...
-				}).fail(function (textStatus, errorThrown) {
-					aDeferred.reject(textStatus, errorThrown);
-				});
-			}
-			return aDeferred.promise();
+	registerPageNavigationEvents() {
+		const self = this,
+			listViewPageDiv = this.getListViewContainer();
+		listViewPageDiv.find('.js-next-page').on('click', (e) => {
+			self.jumpToNextPage(e);
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]:
```suggestion
			this.jumpToNextPage(e);
```

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 7]
```diff
@@ -875,177 +876,119 @@ jQuery.Class("Vtiger_List_Js", {
...
-				jQuery('#pageToJump').val(previousPageNumber);
-				thisInstance.getListViewRecords(urlParams).done(function (data) {
-					thisInstance.updatePagination(previousPageNumber);
-					aDeferred.resolve();
-				}).fail(function (textStatus, errorThrown) {
-					aDeferred.reject(textStatus, errorThrown);
-				});
-			}
+		listViewPageDiv.find('.js-page--previous').on('click', () => {
+			self.jumpToPreviousPage();
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]:
```suggestion
			this.jumpToPreviousPage();
```

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 8]
```diff
@@ -875,177 +876,119 @@ jQuery.Class("Vtiger_List_Js", {
...
-			}
-			var previousPageNumber = parseInt(parseFloat(pageNumber)) - 1;
-			jQuery('#pageNumber').val(previousPageNumber);
-			jQuery('#pageToJump').val(previousPageNumber);
-			thisInstance.getListViewRecords(urlParams).done(function (data) {
-				thisInstance.updatePagination(pageNumber);
-			}).fail(function (textStatus, errorThrown) {
-			});
+		listViewPageDiv.find('.pageNumber').on('click', (e) => {
+			self.jumpToClickedPage($(e.currentTarget));
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]:
```suggestion
			this.jumpToClickedPage($(e.currentTarget));
```

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 9]
```diff
@@ -875,177 +876,119 @@ jQuery.Class("Vtiger_List_Js", {
...
-			var params = thisInstance.getDefaultParams();
-			params.totalCount = -1;
-			params.view = 'Pagination';
-			params.mode = 'getPagination';
-			AppConnector.request(params).done(function (data) {
-				jQuery('.paginationDiv').html(data);
-				thisInstance.registerPageNavigationEvents();
-			});
+		listViewPageDiv.find('.js-count-number-records').on('click', () => {
+			self.updatePaginationAjax(true);
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]:
```suggestion
			this.updatePaginationAjax(true);
```

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 10]
```diff
@@ -875,177 +876,119 @@ jQuery.Class("Vtiger_List_Js", {
...
+			e.stopImmediatePropagation();
+		}).on('keypress', '.js-page-jump', (e) => {
+			self.jumpToPage(e);
+		});
+	},
+	/**
+	 * Jump to next page
+	 */
+	jumpToNextPage(e) {
+		const listViewPageDiv = this.getListViewContainer();
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]: after condition

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 11]
```diff
@@ -875,177 +876,119 @@ jQuery.Class("Vtiger_List_Js", {
...
+		listViewPageDiv.find('.js-page--jump-drop-down').on('click', 'li', (e) => {
+			e.stopImmediatePropagation();
+		}).on('keypress', '.js-page-jump', (e) => {
+			self.jumpToPage(e);
+		});
+	},
+	/**
+	 * Jump to next page
+	 */
+	jumpToNextPage(e) {
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]:
```suggestion
	jumpToNextPage({jQuery} element) {
```

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 12]
```diff
@@ -875,177 +876,119 @@ jQuery.Class("Vtiger_List_Js", {
...
-					element.text(pageCount);
+		listViewPageDiv.find('.js-page--jump-drop-down').on('click', 'li', (e) => {
+			e.stopImmediatePropagation();
+		}).on('keypress', '.js-page-jump', (e) => {
+			self.jumpToPage(e);
+		});
+	},
+	/**
+	 * Jump to next page
+	 */
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]: {jQuery} element

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 13]
```diff
@@ -875,177 +876,119 @@ jQuery.Class("Vtiger_List_Js", {
...
+	},
+	/**
+	 * Jump to clicked page function
+	 * @param {jQuery} element
+	 */
+	jumpToClickedPage(element) {
+		if (element.hasClass('disabled')) {
+			return false;
+		}
+		this.paginationGoToPage(element.data("id"));
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]:
```suggestion
		this.paginationGoToPage(element.data('id'));
```

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 14]
```diff
@@ -875,177 +876,119 @@ jQuery.Class("Vtiger_List_Js", {
...
+	 * @returns {boolean}
+	 */
+	jumpToPage(e) {
+		const self = this,
+			listViewPageDiv = this.getListViewContainer();
+		if (13 === e.which) {
+			e.stopImmediatePropagation();
+			let element = $(e.currentTarget),
+				response = Vtiger_WholeNumberGreaterThanZero_Validator_Js.invokeValidation(element);
+			if (typeof response !== "undefined") {
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]:
```suggestion
			if (typeof response !== 'undefined') {
```

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 15]
```diff
@@ -875,177 +876,119 @@ jQuery.Class("Vtiger_List_Js", {
...
+				element.validationEngine('showPrompt', response, '', "topLeft", true);
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]:
```suggestion
				element.validationEngine('showPrompt', response, '', 'topLeft', true);
```

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 16]
```diff
@@ -875,177 +876,119 @@ jQuery.Class("Vtiger_List_Js", {
...
-						thisInstance.updatePagination(newPageNumber);
-						element.closest('.btn-group ').removeClass('open');
-					}).fail(function (textStatus, errorThrown) {
-					});
+				if (newPageNumber === currentPageNumber) {
+					let params = {
+						text: app.vtranslate('JS_YOU_ARE_IN_PAGE_NUMBER') + " " + newPageNumber,
+						type: 'info'
+					};
+					Vtiger_Helper_Js.showMessage(params);
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]:
```suggestion
					Vtiger_Helper_Js.showMessage(
					{
					...
					});
```

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 17]
```diff
@@ -875,177 +876,117 @@ jQuery.Class("Vtiger_List_Js", {
 			}
 		});
 	},
-	/*
+	/**
+	 * Function to go to page
```
Comment (TomaszQr) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]: @param {type} page

**Pull request merged**